src/main/java/com/zepben/evolve/streaming/get/NetworkConsumerClient.kt

```diff
@@ -128,36 +181,46 @@ class NetworkConsumerClient(
         else if (feeder !is Feeder)
             return GrpcResult.ofError(ClassCastException("Unable to extract feeder network from ${feeder.typeNameAndMRID()}."), false)
 
-        val equipmentResults = getIdentifiedObjects(service, service.getUnresolvedReferenceMrids(Resolvers.equipment(feeder)))
-            .onError { thrown, wasHandled -> return@getFeeder GrpcResult.ofError(thrown, wasHandled) }
+        val mor = MultiObjectResult()
+        mor.objects[feeder.mRID] = feeder
 
-        val mRIDs = service.getUnresolvedReferenceMrids(Resolvers.normalEnergizingSubstation(feeder)).toMutableSet()
+        val result = getEquipmentForContainer(service, mRID).onError { thrown, wasHandled -> return@getFeeder GrpcResult.ofError(thrown, wasHandled) }
+        mor.objects.putAll(result.value.objects)
 
-        feeder.equipment.forEach {
-            if (it is ConductingEquipment) {
-                it.terminals.forEach { terminal ->
-                    mRIDs.addAll(service.getUnresolvedReferenceMrids(Resolvers.connectivityNode(terminal)))
-                }
-            }
+        var res = mor
+        do {
+            val toResolve = getUnresolvedMRIDs(service, res.objects.keys)
+            res = getIdentifiedObjects(service, toResolve).onError { thrown, wasHandled -> return@getFeeder GrpcResult.ofError(thrown, wasHandled) }.value
```

Comment:
Looking at this got me thinking that what if there are no unresolved references. It believe it makes a gRPC request for identified objects under the covers with no mRID's in the request message. I do not know what this does, but we can at least skip making the request by putting a simple check to return the `existing` collection in the following function if there are no `toFetch`

```
private fun processIdentifiedObjects(service: NetworkService, mRIDs: Set<String>): Sequence<ExtractResult> {
        val toFetch = mutableSetOf<String>()
        val existing = mutableSetOf<ExtractResult>()
        mRIDs.forEach { mRID ->  // Only process mRIDs not already present in service
            service.get<IdentifiedObject>(mRID)?.let { existing.add(ExtractResult(it, it.mRID)) } ?: toFetch.add(mRID)
        }

        return stub.getIdentifiedObjects(GetIdentifiedObjectsRequest.newBuilder().addAllMrids(toFetch).build())
            .asSequence()
            .map { extractIdentifiedObject(service, it.identifiedObject) } + existing
    }
```

Reply:
good point, fixed as suggested - also added in a short circuit in case you called processIdentifiedObjects with an empty set of mRIDs.